File: targets/jsts-web/src/vulns/vuln03_reflected_xss.ts
import { Router } from "express";
export const router = Router();
router.get("/vuln03/echo", (req, res) => {
  const q = String(req.query.q || "");
  // VULN:VULN-03:CWE-79:taint reflected XSS: input echoed into HTML unescaped
  res.type("html").send("<html><body><h3>You searched for: " + q + "</h3></body></html>");
});

File: targets/swift/Sources/App/Vuln03ReflectedXss.swift
import Vapor

func registerVuln03(_ app: Application) {
    app.get("vuln03", "echo") { req -> Response in
        let q = req.query[String.self, at: "q"] ?? ""
        // VULN:VULN-03:CWE-79:taint reflected XSS: input echoed into HTML unescaped
        let html = "<html><body><h3>You searched for: \(q)</h3></body></html>"
        return Response(status: .ok, headers: ["Content-Type": "text/html"], body: .init(string: html))
    }
}
